Cesium3DTilesSelection/src/Tileset.cpp

```diff
@@ -339,7 +340,11 @@ Tileset::updateView(const std::vector<ViewState>& frustums, float deltaTime) {
     pOcclusionPool->pruneOcclusionProxyMappings();
   }
 
-  this->_unloadCachedTiles();
+  this->_pTilesetContentManager->tickMainThreadLoading(
+      this->_options.mainThreadLoadingTimeLimit,
+      this->_options);
+
+  this->_unloadCachedTiles(this->_options.tileCacheUnloadTimeLimit);
```

Comment:
Can we call `_unloadCachedTiles` before `tickMainThreadLoading`? It should reduce peak memory usage a little to do it in that order.

Reply:
Good idea! I was initially worried there would be some thrashing if we do that, but I think it's unfounded - since only visited tiles end up on in the finishLoadingQueue and only non-visited tiles are evicted in unloadCachedTiles.